debug-gts.py: support armv9a, add arguments for chunks and extra llvm-mc args

katrinafyi opened the pull request:
addresses #13.

The crash can be investigated with this PR's additional arguments using:
```bash
pipx run scripts/debug-gts.py crash/chase_lev_deque.gts --args='' --debug
```
This tells us the offending opcode is `0x64 0xfc 0xe0 0xc8`, which corresponds to a `casal   x0, x4, [x3]`, which is in v8.1+. I assume llvm-mc defaults to an earlier version of arm64. 

In this PR, I've bumped the default Arm standard to v9a and added a flag to override this if needed. This should be a superset of the standard that aslp supports (v8.6?).

Note to reviewer: I am iffy about the global arguments object. Suggestions are welcome.

katrinafyi committed:
debug-gts.py: support armv9a, add arguments for chunks and extra-args

also tightens up some assertions

l-kent commented:
LLVM defaults to the base ARM64 without any of the additions in later versions, yeah. I figured that having an instruction that wasn't in the base version was probably the cause for the error.

Rather than v9a, it might make sense to just specify v8.6a or whatever our target actually is (have we ever decided on a precise scope?). v9a is apparently a superset of v8.5a but does not include features that were added in later versions of v8 - these have been progressively added with subsequent v9 versions alongside new features - v9.1a is a superset of v8.6a, v9.2a is a superset of v8.7a, and so on.

l-kent commented:
@ailrst Can we get this merged ASAP? Thanks

katrinafyi committed:
use dataclass for command-line arguments.

katrinafyi committed:
use arm v9.3a by default

this is supported on at least llvm 14+, and
should be recent enough that we won't need
another bump in the near future.

katrinafyi reviewed (commented):
✅ I think this argument passing is much better!

katrinafyi commented:
@l-kent  gtirb_semantics-unstable-2024-06-17

Edit: maybe we should make it more resilient to failures too, so we still get partial output in case of errors...

l-kent commented:
Thanks!

Given that you've set it to v9.3a, this particular sort of error shouldn't realistically occur again with any reasonable input, but being able to recover if there is some unknown opcode would still be ideal, yeah.

Although any such opcode would probably cause ASLp to crash too, making this largely irrelevant?
